app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,60 @@
+class Api::V1::BaseController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  before_action :skip_session
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found_handler
+  rescue_from ActiveRecord::RecordNotSaved, with: :not_acceptable_handler
+  rescue_from ActiveRecord::RecordInvalid, with: :not_acceptable_handler
+  rescue_from Api::Errors::PermissionRequired, with: :permission_required_handler
+
+  private
+
+  def authenticate
+    doorkeeper_authorize!
+    super if current_user
+  end
+
+  # @return [User] Current user, or +nil+ if no valid token.
+  def current_user
+    @current_user ||= User.undeleted.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+  end
+
+  # @return [Ordergroup] Current user's ordergroup, or +nil+ if no valid token or user has no ordergroup.
+  def current_ordergroup
+    current_user.try(:ordergroup)
+  end
+
+  def require_ordergroup
+    authenticate
+    current_user.ordergroup or raise Api::Errors::PermissionRequired
+  end
+
+  def skip_session
+    request.session_options[:skip] = true
+  end
+
+  def not_found_handler(e)
+    # remove where-clauses from error message (not suitable for end-users)
+    msg = e.message.try {|m| m.sub(/\s*\[.*?\]\s*$/, '')} || 'Not found'
```

Comment:
is there no cleaner way than this ugly string manipulation?

Reply:
Are there no HTTP status code strings definitions in rails? If not, can we put the hardcoded strings like `Not found` at least on top of the file as const defines?